ContractExamples/contracts/alert/src/lib.rs

```diff
@@ -0,0 +1,54 @@
+#![no_std]
+use soroban_sdk::{contract, contractimpl, contracttype, symbol_short, Env, Symbol, Map, String};
+
+const ALERTS: Symbol = symbol_short!("ALERTS");
+const VIOLATION: Symbol = symbol_short!("VIOLATION");
+
+#[contract]
+pub struct Alert;
+
+#[contracttype]
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[repr(u32)]
+pub enum MonitorAnalysisStatus {
+    NoViolation = 0,
+    Violation = 1,
+    Unknown = 2
+}
+
+type AlertMap = Map<String, MonitorAnalysisStatus>;
+
+/*
+    The Alert contract gets called by the monitor executor, whenever a transaction gets analyzed.
+    If the monitor detected a property violation, then the Alert contract emits a warning.
+
+    For each tx hash, for which alert was called, stores MonitorAnalysisStatus, the value of which
+    depends on whether the monitor detected a violation for that transaction.
+*/
+#[contractimpl]
+impl Alert {
+
+    pub fn emit_warning_if_violation(env: Env, tx_hash: String, status: MonitorAnalysisStatus) -> MonitorAnalysisStatus {
+        // Get the current alerts
+        let mut alerts: AlertMap = env.storage().instance().get(&ALERTS).unwrap_or(AlertMap::new(&env));
+
+        // Add to history and save
+        alerts.set(tx_hash, status);
+        env.storage().instance().set(&ALERTS, &alerts);
+
+        // The contract instance will be bumped to have a lifetime of at least 100 ledgers if the current expiration lifetime at most 50.
+        // If the lifetime is already more than 100 ledgers, this is a no-op. Otherwise,
+        // the lifetime is extended to 100 ledgers. This lifetime bump includes the contract
+        // instance itself and all entries in storage().instance().
+        env.storage().instance().extend_ttl(50, 100);
```

Comment:
I think this was a hack for Igor's test contract. If you use `instance`, the storage TTL will be the contract's TTL, so I don't think it needs to be bumped.

Reply:
Why? Persistent gives an individual TTL to every field.

Reply:
My understanding is that `Instance` still gives a finite TTL, so you do need to periodically extend it.

Reply:
Actually, reading [this](https://developers.stellar.org/docs/smart-contracts/guides/storage/use-instance) makes me even more confused as to why `.extend_ttl` even exists/what it does. If "Under the hood, instance storage is exactly like persistent storage. The only difference is that anything stored in instance storage has an archival TTL that is tied to the contract instance itself. So, if a contract is live and available, the instance storage is guaranteed to be so, too" is true, you would never need to actually `extend_ttl`

Reply:
Verbatim from the section you linked: `For Temporary and Persistent storage, each entry has its own TTL and must be extended individually.`

Reply:
No, I mean what is the point of `.instance().extend_ttl(...)` given the above.

Reply:
You're probably right that `Instance` is the best choice for this sort of contract regardless.